### PR TITLE
docs mostly

### PR DESCRIPTION
**Pull request by artofnothingness**
closes this:
- Explain roles of State and ControlSequence in doxygen / comments in code / naming. No need for README documentation, just make clear in code
-  Reduce copies in the optimizer class, if any
-  Remove, where reasonable / practical auto from method definitions: getGlobalPlanConsideringBounds potentially getControlFromSequence (only auto returning function in the optimizer)

### CODE REVIEW

File: include/mppic/tensor_wrappers/state.hpp [hunk 1]
```diff
@@ -70,6 +73,10 @@ class StateIdxes
   unsigned int dim_{0};
 };
 
+/**
+ * struct State - combined data with it's named indexes of last dimension variables
```
Comment (SteveMacenski) [inline include/mppic/tensor_wrappers/state.hpp]: These comments don't really explain how these are used in the code. What is this struct containing mean to the algorithm?

File: include/mppic/tensor_wrappers/control_sequence.hpp [hunk 2]
```diff
@@ -42,6 +45,9 @@ class ControlSequnceIdxes
   unsigned int dim_{0};
 };
 
+/**
+ * struct ControlSequence - combined data with it's named indexes of last dimension variables
```
Comment (SteveMacenski) [inline include/mppic/tensor_wrappers/control_sequence.hpp]: These comments don't really explain how these are used in the code. What is this struct containing mean to the algorithm?

**Review by SteveMacenski — COMMENTED** [review]
Careful, a bunch of places in this PR things with 2 spaces were replaced with 4 spaces for indentation, making it inconsistent. 

It also changes having the function definitions have the `{` on the same line as the function definition, not on a new line as is typical for coding style

File: include/mppic/tensor_wrappers/control_sequence.hpp (hunk 2 above)
Comment (artofnothingness) [reply include/mppic/tensor_wrappers/control_sequence.hpp]: Have tried to make it more clear

File: include/mppic/tensor_wrappers/state.hpp (hunk 1 above)
Comment (artofnothingness) [reply include/mppic/tensor_wrappers/state.hpp]: Have tried to make it more clear

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**